bigchaindb/web/websocket_server.py

```diff
@@ -0,0 +1,174 @@
+# modules. For now, I prefer to keep everything in this module. Why? Because
+# those functions are needed only here.
+#
+# When we will extend this part of the project and we find that we need those
+# functionalities elsewhere, we can start creating new modules and organizing
+# things in a better way.
+
+
+import json
+import asyncio
+import logging
+import threading
+from uuid import uuid4
+
+import aiohttp
+from aiohttp import web
+
+from bigchaindb import config
+from bigchaindb.events import EventTypes
+
+
+logger = logging.getLogger(__name__)
+POISON_PILL = 'POISON_PILL'
+EVENTS_ENDPOINT = '/api/v1/streams/valid_tx'
+
+
+def _put_into_capped_queue(queue, value):
+    """Put a new item in a capped queue.
+
+    If the queue reached its limit, get the first element
+    ready and put the new one. Note that the first element
+    will be lost (that's the purpose of a capped queue).
+
+    Args:
+        queue: a queue
+        value: the value to put
+    """
+    while True:
+        try:
+            queue.put_nowait(value)
+        except asyncio.QueueFull:
+            queue.get_nowait()
+        else:
+            return
+
+
+def _multiprocessing_to_asyncio(in_queue, out_queue, loop):
+    """Bridge between a synchronous multiprocessing queue
+    and an asynchronous asyncio queue.
+
+    Args:
+        in_queue (multiprocessing.Queue): input queue
+        out_queue (asyncio.Queue): output queue
+    """
+
+    while True:
+        value = in_queue.get()
+        loop.call_soon_threadsafe(_put_into_capped_queue, out_queue, value)
+
+
+class Dispatcher:
+    """Dispatch events to websockets.
+
+    This class implements a simple publish/subscribe pattern.
+    """
+
+    def __init__(self, event_source):
+        """Create a new instance.
+
+        Args:
+            event_source: a source of events. Elements in the queue
+            should be strings.
+        """
+
+        self.event_source = event_source
+        self.subscribers = {}
+
+    def subscribe(self, uuid, websocket):
+        """Add a websocket to the list of subscribers.
+
+        Args:
+            uuid (str): a unique identifier for the websocket.
+            websocket: the websocket to publish information.
+        """
+
+        self.subscribers[uuid] = websocket
+
+    @asyncio.coroutine
+    def publish(self):
+        """Publish new events to the subscribers."""
+
+        while True:
+            event = yield from self.event_source.get()
+            str_buffer = []
+
+            if event == POISON_PILL:
+                return
+
+            if isinstance(event, str):
+                str_buffer.append(event)
+
+            elif event.type == EventTypes.BLOCK_VALID:
+                block = event.data
+
+                for tx in block['block']['transactions']:
+                    asset_id = tx['id'] if tx['operation'] == 'CREATE' else tx['asset']['id']
+                    data = {'blockid': block['id'],
```

Comment:
In broader terms, I see that as an API design specification question, and it has already been answered for the HTTP API, and in some way also for the websocket API. It is too late right now to change it I think.